deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/CustomLayer/ElementWiseMultiplicationLayer_impl.java

```diff
@@ -0,0 +1,160 @@
+
+//        INDArray weightGrad = Nd4j.zeros( gradientViews.get(ElementWiseParamInitializer.WEIGHT_KEY).shape(),'f'); //f order
+        INDArray weightGrad =  gradientViews.get(ElementWiseParamInitializer.WEIGHT_KEY); //f order
+//        reset weight gradients
+        weightGrad.subi(weightGrad);
+
+        for(int row =0;row<input.rows();row++){
+            weightGrad.addi(input.getRow(row).mul(delta.getRow(row)));
+        }
+
+//        weightGrad.addi(input.getRow(0).mul(delta));
+
+        INDArray biasGrad = gradientViews.get(ElementWiseParamInitializer.BIAS_KEY);
+        delta.sum(biasGrad, 0); //biasGrad is initialized/zeroed first
+
+        ret.gradientForVariable().put(ElementWiseParamInitializer.WEIGHT_KEY, weightGrad);
+        ret.gradientForVariable().put(ElementWiseParamInitializer.BIAS_KEY, biasGrad);
+
+        INDArray epsilonNext = Nd4j.zeros(params.get(ElementWiseParamInitializer.WEIGHT_KEY).shape());
+
+        for(int row =0;row<delta.rows();row++){
+            epsilonNext.addi(params.get(ElementWiseParamInitializer.WEIGHT_KEY).mul(delta.getRow(row)));
+        }
+
+        return new Pair<>(ret, epsilonNext);
+    }
+
+/*    @Override
+    public void fit(INDArray input) {}*/
+
+    /**
+     * Returns true if the layer can be trained in an unsupervised/pretrain manner (VAE, RBMs etc)
+     *
+     * @return true if the layer can be pretrained (using fit(INDArray), false otherwise
+     */
+    @Override
+    public boolean isPretrainLayer() {
+        return false;
+    }
+
+    public INDArray preOutput(boolean training) {
+        applyDropOutIfNecessary(training);
+        INDArray b = getParam(DefaultParamInitializer.BIAS_KEY);
+        INDArray W = getParam(DefaultParamInitializer.WEIGHT_KEY);
+
+        //Input validation:
+        if ( input.columns() != W.columns()) {
+
+            throw new DL4JInvalidInputException(
+                    "Input size (" + input.columns() + " columns; shape = " + Arrays.toString(input.shape())
+                            + ") is invalid: does not match layer input size (layer # inputs = "
+                            + W.shapeInfoToString() + ") " + layerId());
+        }
+
+        if (conf.isUseDropConnect() && training && layerConf().getDropOut() > 0) {
+            W = Dropout.applyDropConnect(this, DefaultParamInitializer.WEIGHT_KEY);
+        }
+
+//        System.out.println(W.shapeInfoToString());
+//        System.out.println(input.shapeInfoToString());
+
+        INDArray ret = Nd4j.zeros(input.rows(),input.columns());
+
+        for(int row = 0; row<input.rows();row++){
+            ret.put(new INDArrayIndex[]{NDArrayIndex.point(row), NDArrayIndex.all()},input.getRow(row).mul(W).addRowVector(b));
+        }
+
+//        INDArray ret = input.mul(W).addRowVector(b);
+
+        if (maskArray != null) {
+            applyMask(ret);
+        }
+
+        return ret;
+    }
+
+
+    @Override
+    public INDArray activationMean() {
```

Comment:
This method does not exist anymore on Layer API:
https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/api/Layer.java